lib/ansible/plugins/lookup/__init__.py

```diff
@@ -112,8 +113,11 @@ def find_file_in_search_path(self, myvars, subdir, needle, ignore_missing=False)
         else:
             paths = self.get_basedir(myvars)
 
-        result = self._loader.path_dwim_relative_stack(paths, subdir, needle)
-        if result is None and not ignore_missing:
-            self._display.warning("Unable to find '%s' in expected paths." % needle)
+        result = None
```

Comment:
[just a note inspired by branch, not something that needs to be addressed by this pr]

Looking though code that uses find_file_in_search_path(), there are some lookup plugins that dont really
check the return value (ini looks like it can throw exceptions if result is None, csvfile doesnt check for !None, but does at least try/except around open() it)